### DISCUSSION TITLE
Shared Links with Random API?

### DISCUSSION THREAD

**Discussion by rabenimmermehr** [bpatrik/pigallery2#354] [Q&A] [answered]
Hi there!

Is there an option to combine the "share" feature with the "random" feature?

What I'm looking for is an option to share a single link displaying a random image without the viewer having to log in :) 

Thank you for your work, as others mentioned, this app hits the sweetspot of features: Not too much and not too little!

Cheers,
Rabe

**Comment by bpatrik** [comment]
For the Random API you can do a full search query so you can create more or less an arbitrary set to display, while sharing can only share a single folder.

Alternatively you can also disable passwords on shared links.

**Comment by rabenimmermehr** [comment, marked as answer]
If I understood that correctly, my request isn't feasible, because the random feature could theoretically display images from outside shared folders, thereby circumventing password protection.

In my case, I now solved it by disabling passwords in general by setting the environement variable ˋClient-authenticationRequired=falseˋ.

Anyone in my home network can now call the random link and get a picture :)
